packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeRenderer.tsx

```diff
@@ -46,20 +45,57 @@ export function TreeRenderer({ rootNodes, expandNode, localizedStrings, selectNo
     selectionMode: selectionMode ?? "single",
   });
 
+  const flatNodes = useMemo(() => flattenNodes(rootNodes), [rootNodes]);
+
   return (
     <LocalizationContextProvider localizedStrings={localizedStrings}>
       <Tree.Root style={{ height: "100%", width: "100%" }}>
-        <TreeLevelRenderer
-          {...treeProps}
-          nodes={rootNodes}
-          expandNode={expandNode}
-          onNodeClick={onNodeClick}
-          onNodeKeyDown={onNodeKeyDown}
-          isNodeSelected={isNodeSelected}
-        />
+        {flatNodes.map((flatNode) => (
+          <TreeNodeRenderer
+            {...treeProps}
+            expandNode={expandNode}
+            onNodeClick={onNodeClick}
+            onNodeKeyDown={onNodeKeyDown}
+            node={flatNode}
+            key={flatNode.id}
+            selected={isNodeSelected?.(flatNode.id)}
```

Comment:
Is there a reason why `TreeNodeRenderer` takes `selected` flag instead of `isNodeSelected`?

Reply:
selected comes from itwinUI side, we could pass isNodeSelected and omit selected, then call it inside the `TreeNodeRenderer`, but I don't think  that's necessary as it will just add additional code lines we can avoid.

Reply:
I think it makes sense handling `isNodeSelected` inside `TreeNodeRenderer` because at the moment we call `isNodeSelected` even for error nodes. Also looking into this more I think `TreeNodeRenderer` should not allow passing `expanded` flag as it is not used internally.